In [0]:
from sklearn.datasets import load_iris

In [0]:
iris=load_iris()

In [0]:
X=iris.data

In [0]:
y=iris.target

In [5]:
y

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

In [9]:
y.dtype

dtype('int64')

# Train Test Split

In [0]:
from sklearn.model_selection import train_test_split

In [0]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)

In [0]:
import tensorflow as tf

# **Contrib.learn**
Let's see how to use the simpler contrib.learn interface!

In [0]:
import tensorflow.contrib.learn.python

from tensorflow.contrib.learn.python import learn as learn

There are several high level abstraction calls to models in learn, you can explore them with Tab, but we will use DNNClassifier, which stands for Deep Neural Network:

In [0]:
feature_columns = tf.contrib.learn.infer_real_valued_columns_from_input( X_train)
classifier=learn.DNNClassifier(feature_columns=feature_columns,hidden_units=[10,20,10],n_classes=3)

In [0]:
classifier.fit(X_train, y_train, steps=200, batch_size=32)

In [41]:
predictions = list(classifier.predict(X_test, as_iterable=True))

INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from /tmp/tmp83mf2h68/model.ckpt-200
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.


In [0]:
from sklearn.metrics import classification_report,confusion_matrix

In [43]:
print(classification_report(y_test,predictions))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        13
           1       0.74      0.93      0.82        15
           2       0.92      0.71      0.80        17

    accuracy                           0.87        45
   macro avg       0.89      0.88      0.87        45
weighted avg       0.88      0.87      0.87        45



We can play around with hidden_units, steps and batch size!

In [0]:
feature_columns = tf.contrib.learn.infer_real_valued_columns_from_input( X_train)
classifier=learn.DNNClassifier(feature_columns=feature_columns,hidden_units=[20,20,20],n_classes=3)
classifier.fit(X_train, y_train, steps=300, batch_size=32)
predictions = list(classifier.predict(X_test, as_iterable=True))

In [47]:
print(classification_report(y_test,predictions))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        13
           1       1.00      0.87      0.93        15
           2       0.89      1.00      0.94        17

    accuracy                           0.96        45
   macro avg       0.96      0.96      0.96        45
weighted avg       0.96      0.96      0.96        45



**So now we can clearly see the increase in the Accuracy from .87 to .96**